### PR TITLE
intel_alm: DSP inference

### PR DESCRIPTION
**Pull request by Ravenslofty**
This adds basic support for the multiply-accumulate slices within the Cyclone V, using `mul2dsp.v` to map them. They're rather unusual in that a slice can be three 9x9 multipliers, two 18x18 multipliers or a 27x27 multiplier.  Ideally this would have a Yosys pass to attempt to pack adders and flops into the cell, but this works fine as a minimum viable product.

cc @mwkmwkmwk

### CODE REVIEW

File: techlibs/intel_alm/common/quartus_rename.v [hunk 1]
```diff
@@ -119,3 +121,54 @@ module MISTRAL_MLAB(input [4:0] A1ADDR, input A1DATA, A1EN, CLK1, input [4:0] B1
 );
 
 endmodule
+
+
+module MISTRAL_MUL27X27(A, B, Y);
+
+parameter A_SIGNED = 0;
```
Comment (mwkmwkmwk) [inline techlibs/intel_alm/common/quartus_rename.v:128]: why do these cells have parameters here but not in `dsp_sim.v`?  IMO these should be dropped

File: techlibs/intel_alm/common/quartus_rename.v [hunk 2]
```diff
@@ -119,3 +121,54 @@ module MISTRAL_MLAB(input [4:0] A1ADDR, input A1DATA, A1EN, CLK1, input [4:0] B1
...
+
+endmodule
+
+
+module MISTRAL_MUL18X18(A, B, Y);
+
+parameter A_SIGNED = 0;
+parameter B_SIGNED = 1;
+parameter A_WIDTH = 18;
+parameter B_WIDTH = 19;
```
Comment (mwkmwkmwk) [inline techlibs/intel_alm/common/quartus_rename.v:148]: this doesn't match `dsp_sim.v`

File: techlibs/intel_alm/common/dff_sim.v [hunk 3]
```diff
@@ -61,12 +61,14 @@ module MISTRAL_FF(
 `ifdef cyclonev
 specify
     (posedge CLK => (Q : DATAIN)) = 262;
+    (posedge CLK => (Q : SDATA)) = 262;
```
Comment (mwkmwkmwk) [inline techlibs/intel_alm/common/dff_sim.v:64]: please do not sneak unrelated changes like that

File: techlibs/intel_alm/common/quartus_rename.v (hunk 1 above)
Comment (Ravenslofty) [reply techlibs/intel_alm/common/quartus_rename.v:128]: I dropped them, yeah. Not entirely sure what I was thinking.

File: techlibs/intel_alm/common/megafunction_bb.v [hunk 4]
```diff
@@ -129,3 +129,17 @@ output [data_width-1:0] portbdataout;
 input ena0, clk0, clk1;
 
 endmodule
+
```
Comment (mwkmwkmwk) [inline techlibs/intel_alm/common/megafunction_bb.v:132]: shouldn't there be a cyclone10gx_mac as well?
